src/provider/provider_tracking.c

```diff
@@ -7,43 +7,81 @@
  *
  */
 
-#include "provider_tracking.h"
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <umf/memory_pool.h>
+#include <umf/memory_provider.h>
+#include <umf/memory_provider_ops.h>
+
 #include "base_alloc_global.h"
 #include "critnib.h"
 #include "ipc_cache.h"
 #include "ipc_internal.h"
+#include "provider_tracking.h"
 #include "utils_common.h"
 #include "utils_concurrency.h"
 #include "utils_log.h"
 
-#include <umf/memory_pool.h>
-#include <umf/memory_provider.h>
-#include <umf/memory_provider_ops.h>
-
-#include <assert.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+#define MAX_LEVELS_OF_MAPS 8
 
 uint64_t IPC_HANDLE_ID = 0;
 
 struct umf_memory_tracker_t {
     umf_ba_pool_t *alloc_info_allocator;
-    critnib *alloc_segments_map;
+    critnib *alloc_segments_map[MAX_LEVELS_OF_MAPS];
```

Comment:
Could you please add a comment describing why we need these multi-level maps? We should say something like "Multilevel maps are needed to support the case when one memory pool acts as a memory provider for another memory pool (nested memory pooling)"

Reply:
@vinser52 Done